### DISCUSSION TITLE
Would love to know how this library works!

### DISCUSSION THREAD

**Discussion by smblee** [paoloricciuti/sveltekit-search-params#39] [General]
Ran across this repo from This week in svelte youtube video and would love to get a high level breakdown of how this library works. Really neat concept to make query params reactive!

**Comment by paoloricciuti** [comment]
Hi, this is actually pretty simple: the value of the store is derived from `$page.url.searchParams` and when you update it the custom `set` function get's called that instead of setting the actual value just navigate to the correct url, this in turn update $page that in turn update the value. And that's basically it.

All the rest is just to deal with small edge cases and to make typescript happy.

If you have any specific questions feel free to ask.
